File: backend/app/services/user_service.py
from sqlalchemy.orm import Session

from app.models.user import User
from app.repositories.user_repository import UserRepository
from app.schemas.user import UserUpdate


class UserService:
    def __init__(self, db: Session) -> None:
        self.users = UserRepository(db)

    def update(self, user: User, data: UserUpdate) -> User:
        return self.users.update(
            user,
            name=data.name,
            monthly_salary=data.monthly_salary,
        )

    def delete(self, user: User) -> None:
        self.users.delete(user)

File: frontend/src/services/user.service.ts
import { api } from "@/lib/api";
import type { User, UserUpdate } from "@/types/user";

export const userService = {
  async me(): Promise<User> {
    const { data } = await api.get<User>("/users/me");
    return data;
  },

  async update(payload: UserUpdate): Promise<User> {
    const { data } = await api.patch<User>("/users/me", payload);
    return data;
  },
};
